packages/block-library/src/post-author/edit.js

```diff
@@ -21,32 +19,39 @@ import {
 	withFontSizes,
 } from '@wordpress/block-editor';
 import { PanelBody, SelectControl, ToggleControl } from '@wordpress/components';
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
 const DEFAULT_AVATAR_SIZE = 24;
 
-async function getAuthorDetails( authorId, setAuthorDetails ) {
-	const authorDetails = await apiFetch( {
-		path: '/wp/v2/users/' + authorId + '?context=edit',
-	} );
-	setAuthorDetails( authorDetails );
-}
-
-function PostAuthorDisplay( {
-	postAuthor,
-	setPostAuthor,
-	authorDetails,
-	setAuthorDetails,
-	props,
+function PostAuthorEdit( {
+	isSelected,
+	fontSize,
+	setFontSize,
+	context,
+	attributes,
+	setAttributes,
 } ) {
-	const ref = useRef();
+	const { postType, postId } = context;
+
+	const { authorId, authorDetails, authors } = useSelect( ( select ) => {
+		const { getEditedEntityRecord, getUser, getAuthors } = select( 'core' );
+		const { author: _authorId } = getEditedEntityRecord(
+			'postType',
+			postType,
+			postId
+		);
```

Comment:
There's an error here, but not one which would be encountered in practice because we preload post data for the edited post. `getEditedEntityRecord` may return `null` or `undefined`, so it's not safe to destructure like this.

```suggestion
		const _authorId = getEditedEntityRecord(
			'postType',
			postType,
			postId
		)?.author;
```

I do wish these issues would be easier to detect. I think it could come from one or a combination of:

- Type-checking would highlight the error in knowing the return value of `getEditedEntityRecord` as being possible nullish (#18838)
- Disable REST preloading for the end-to-end tests. This would highlight the bugs, but have the effect that tests may run slower, and diverge from how a real user would experience the editor.